### PR TITLE
Templates design updates

### PR DESCRIPTION
**Pull request by AntoineAugusti**
Make several design changes when creating, editing or sending templates.

## Main changes
- Content changes
- Add more context
- Make it easier to send a test message to yourself
- Clarify how bulk sending works
- Update how we display the scheduler

I'll let @e-pang and @anikbrazeau focus on UI changes and @jimleroyer you can take a quick look at the code!

## Links
- [Figma](https://www.figma.com/file/fRVRKm7cM8MUuBWUJbrfgU/Sending-from-web-interface-%7C-Envoyer-par-interface-web?node-id=2803%3A3756)
- Trello: https://trello.com/c/8HXeIRkx/506-create-a-new-template-update-button-copy-and-hierarchy-create-template-button-goes-to-a-new-screen and https://trello.com/c/N5UK9MXt/507-send-message-update-send-to-add-recipients-add-subheading-description-text-update-positioning-and-copy-of-buttons-and-links

### CODE REVIEW

File: app/assets/javascripts/scheduler/SetDateTime/SetDateTime.js [hunk 1]
```diff
@@ -18,46 +18,36 @@ const setDateAndTimeValue = val => {
   }
 };
```
Comment (AntoineAugusti) [inline app/assets/javascripts/scheduler/SetDateTime/SetDateTime.js:20]: Update how the scheduler is displayed. Previously it was appearing when selecting radio buttons, now you can make it appear when clicking a button. If the scheduler is shown and you click "send now", make sure we actually clear the scheduled time

File: app/templates/partials/nav/gc_header.html [hunk 2]
```diff
@@ -51,7 +51,7 @@
     </nav>
     <!-- end sub nav -->
   </div>
-  <div>
+  <div class="pt-5 pb-6">
```
Comment (AntoineAugusti) [inline app/templates/partials/nav/gc_header.html:54]: More vertical space around GC Notify or GC Notify + service name

**Review by e-pang — CHANGES_REQUESTED** [review]
Thanks Antoine, the updates are looking good. 
Just this one update:
- After clicking “Send test message to yourself” the back button goes to the personalise screen instead of back to the previous “Ready to send” screen. Can you update?

File: app/translations/csv/fr.csv [hunk 3]
```diff
@@ -1388,4 +1391,17 @@
 "The email was rejected because of its attachments","Le courriel a été rejeté à cause des pièces jointes"
 "Opens in a new tab","S’ouvre dans un nouvel onglet"
 "For added security, GC Notify has sent you an email message with a security code to confirm you still control a valid Government email address.","Pour plus de sécurité, GC Notification vous a envoyé un courriel avec un code de sécurité pour confirmer que vous contrôlez toujours une adresse courriel gouvernementale valide."
+"Ready to send?","Prêt à envoyer?"
+"Add recipients","Ajouter des destinataires"
+"Edit this template","Modifier le gabarit"
```
Comment (anikbrazeau) [inline app/translations/csv/fr.csv:1396]: The heading when you click in French becomes "Modifier gabarit" ... is there a way to fix that?

File: app/templates/views/send.html [hunk 4]
```diff
@@ -2,28 +2,39 @@
...
   }}
 
+  <p class="mb-0">
+    {{ _('Save your file as a') }} {% set csv_acronym = _('Comma Separated Values') %}
+    {% set tsv_acr = _('Tab Separated Values') %} {% set ods_acronym = _('Open Document Spreadsheet') %}
+    <acronym title="{{ csv_acronym }}">{{ _('CSV') }}</acronym>,
+    <acronym title="{{ tsv_acronym }}">{{ _('TSV') }}</acronym>,
+    <acronym title="{{ ods_acronym }}">{{ _('ODS') }}</acronym> {{ _('or Microsoft Excel spreadsheet') }}
+  </p>
+
```
Comment (anikbrazeau) [inline app/templates/views/send.html:29]:
```suggestion
```

File: app/templates/views/send.html [hunk 5]
```diff
@@ -2,28 +2,39 @@
...
   <div class="page-footer mb-12 clear-both contain-floats">
-    {% set txt = _('Choose a file') %} {{file_upload( form.file, button_text=txt
-    )}}
+    {{ file_upload( form.file, button_text=_('Choose a file')) }}
   </div>
```
Comment (anikbrazeau) [inline app/templates/views/send.html:33]:
```suggestion
  <p class="mb-0">
    {{ _('Save your file as a') }} {% set csv_acronym = _('Comma Separated Values') %}
    {% set tsv_acr = _('Tab Separated Values') %} {% set ods_acronym = _('Open Document Spreadsheet') %}
    <acronym title="{{ csv_acronym }}">{{ _('CSV') }}</acronym>,
    <acronym title="{{ tsv_acronym }}">{{ _('TSV') }}</acronym>,
    <acronym title="{{ ods_acronym }}">{{ _('ODS') }}</acronym> {{ _('or Microsoft Excel spreadsheet') }}
  </p>

```

File: app/templates/views/check/ok.html [hunk 6]
```diff
@@ -4,32 +4,30 @@
...
+      'default',
+      with_tick=True
+    )
+  }}
 
   {% if not request.args.from_test %}
   <h2 class="heading-medium">{{ _('Recipients') }}</h2>
-  <h3 class="file-name" id="{{ file_contents_header_id }}">{{ original_file_name }}</h3>
+  <p>
+    {{ _("{} recipients from {}.").format(count_of_recipients, original_file_name) }}
```
[the remark this answers is not in the document]
Comment (anikbrazeau) [reply app/templates/views/check/ok.html:29]: Also seems to be 2 missing FR strings here?
Screen Shot 2021-05-19 at 4 15 13 PM

"Review before sending","Réviser avant l'envoi"
"was uploaded.", "a été téléversé"

File: app/templates/views/check/ok.html [hunk 7]
```diff
@@ -64,48 +58,44 @@ <h3 class="file-name" id="{{ file_contents_header_id }}">{{ original_file_name }
...
+    </p>
+  {% endif %}
+
   {% endif %}
 
 
+  <h2 class="heading-medium">{{ _('Message preview') }}</h2>
+
+  <p>
+    {{ _("The example below previews row {} from {}.").format(preview_row, original_file_name) }}
```
Comment (anikbrazeau) [inline app/templates/views/check/ok.html:73]:
```suggestion
    {{ _("The example below previews one recipient (row {} from {}.").format(preview_row, original_file_name) }})
```

File: app/translations/csv/fr.csv [hunk 8]
```diff
@@ -1388,4 +1391,17 @@
...
+"Send test message to yourself","Envoyer un message test à soi-même"
+"'{}' template saved","Gabarit '{}' enregistré"
+"Your file must include columns with recipients and variables from your personalised message.","Votre fichier doit inclure des colonnes de destinataires et les variables comprises dans votre message personnalisé."
+"Message preview","Aperçu du message"
+"The example below previews row {} from {}.","L’exemple ci-dessous ne montre que la ligne {} de {}."
```
Comment (anikbrazeau) [inline app/translations/csv/fr.csv:1401]:
```suggestion
"The example below previews one recipient (row {} from {}).","L’exemple ci-dessous ne montre qu'un destinataire (de la ligne {} de {})."
```

File: app/templates/views/check/ok.html (hunk 6 above)
Comment (anikbrazeau) [inline app/templates/views/check/ok.html:29]: can this be singular for 1 recipient?

File: app/translations/csv/fr.csv [hunk 9]
```diff
@@ -1388,4 +1391,17 @@
...
+"{} recipients from {}.","{} destinataires provenant de {}."
```
Comment (anikbrazeau) [inline app/translations/csv/fr.csv:1402]: can this be singular for 1 recipient?

File: app/templates/partials/csv-send.html [hunk 10]
```diff
@@ -1,26 +1,21 @@
...
-  {% set type_2 = _('Tab Separated Values') %} {% set type_3 = _('Open Document Spreadsheet') %}
-  <acronym title="type_1">{{ _('CSV') }}</acronym>,
-  <acronym title="type_2">{{ _('TSV') }}</acronym>,
-  <acronym title="type_3">{{ _('ODS') }}</acronym> {{ _('or Microsoft Excel spreadsheet') }}
+<p>
+  {{ _("For example:") }}
 </p>
 
 <div class="spreadsheet" data-module="fullscreen-table">
-  {% set hint= _('Example') %} {% call(item, row_number) list_table( example,
```
Comment (anikbrazeau) [inline app/templates/partials/csv-send.html:16]: Is there a way to translate the example in French? 
"email address" =  "adresse courriel"
"phone number" = "numéro de téléphone"
"example" = "exemple"
Screen Shot 2021-05-19 at 4 23 48 PM

**Comment by AntoineAugusti** [comment]
@e-pang @anikbrazeau Thanks for your review! Addressed your feedback. Can you check one more time that things are good? @jimleroyer will check stuff on the code side just after.

**Review by e-pang — APPROVED** [review]
Looks good to me now, thanks for updating Antoine!

File: requirements-app.txt [hunk 11]
```diff
@@ -38,6 +38,6 @@ awscli-cwlogs>=1.4.6,<1.5
 # Putting upgrade on hold due to v1.0.0 using sha512 instead of sha1 by default
 itsdangerous==0.24  # pyup: <1.0.0
 
-git+https://github.com/cds-snc/notifier-utils.git@43.9.1#egg=notifications-utils
+git+https://github.com/cds-snc/notifier-utils.git@789b96d8e015ba7df951723ddb227c03b6d331cc#egg=notifications-utils
```
Comment (jimleroyer) [inline requirements-app.txt:41]: Do we want to use a tag here instead of the commit SHA ID?

**Review by jimleroyer — APPROVED** [review]
It's one of these big PR where one is tempted to blindly approve. There are  many kind of changes too which does not make this easy. I tried to catch obvious things but it looks good overall.

I left a few comments, some reflecting and others that could be addressed here or later (e.g. using sha id instead of labeled version).

There is a lot in there. This is great work and will have a good impact.

**Comment by AntoineAugusti** [comment]
@jimleroyer Thanks for the review!

About the scheduler, I think it's out of scope for this PR. I didn't change a lot of things compared to what we have at the moment and I don't want to start _now_ a refactor of this. I do agree that it could be cleaned up! I'll log a "technical debt" card.

For the utils dependency, I used a commit sha for a short bit on this PR only because it was not merged/tagged just yet and I wanted to be able to pull it in the review app. That was a hack while this PR was in review. I've merged/tagged utils and used a tag before merging 👍

**Pull request merged**